### PR TITLE
Avoid calling global_vars_set in constant_typing when not in section

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
@coqbot bench

File: kernel/constant_typing.ml [hunk 1]
```diff
@@ -31,6 +31,11 @@ module NamedDecl = Context.Named.Declaration
 let check_section_variables env declared_vars body typ =
   let env_ids = ids_of_named_context_val (named_context_val env) in
   Id.Set.iter (fun id -> if not (Id.Set.mem id env_ids) then Type_errors.error_unbound_var env id) declared_vars;
+  if List.is_empty (named_context env) then begin
+    assert (Id.Set.is_empty declared_vars);
```
Comment (ppedrot) [inline kernel/constant_typing.ml:35]: The same error as below rather than an anomaly might be better for robustness against bad input.

Comment (SkySkimmer) [reply kernel/constant_typing.ml:35]: we could also say that the assert is not necessary since we checked in the iter just above

**Comment by ppedrot** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**